### PR TITLE
Removed market research opt out from onboarding flow

### PR DESCRIPTION
**Pull request by liamness**
## What does this change?

The removes the marketing research opt out from the onboarding flow, and also any related code.

## How to test

- Go to `/register` and enter an email address,
- Visit the sign in link sent to that address (locally, I have been logging the URL to the terminal from the running IDAPI process)
- Enter a password
- Observe that the market research opt out is no longer visible on the `/consents/communication` page
- Continue through the onboarding flow
- Observe that no information relating to market research consent is visible on the final `/consents/review` page

## How can we measure success?

This change has already been tested previously.

## Have we considered potential risks?

No major risks anticipated.

### Before

Screenshot_2021-12-08_at_12 14 28

### After

image

## Accessibility

- [ ] [Tested with screen reader](https://guardian.github.io/source/?path=/docs/docs-06-accessibility--page#screen-readers)
- [ ] [Navigable with keyboard](https://guardian.github.io/source/?path=/docs/docs-06-accessibility--page#keyboard-navigation)
- [ ] [Colour contrast passed](https://guardian.github.io/source/?path=/docs/docs-06-accessibility--page#colour-contrast)
- [ ] [The change doesn't use only colour to convey meaning](https://guardian.github.io/source/?path=/docs/docs-06-accessibility--page#use-of-colour)

### CODE REVIEW

File: src/server/lib/middleware/login.ts [hunk 1]
```diff
@@ -42,6 +42,7 @@ export const loginMiddleware = async (
   const sc_gu_la = req.cookies.SC_GU_LA;
 
   if (!sc_gu_u || !sc_gu_la) {
+    console.log('DOIN A REDIRECT');
```
Comment (ob6160) [inline src/server/lib/middleware/login.ts:45]: might steal this log message for my own use haha

**Review by ob6160 — APPROVED** [review]
lgtm! congrats on your first gateway pr 🥳

**Comment by liamness** [comment]
I think there are changes that need reviewing in Chromatic: https://www.chromatic.com/build?appId=60929d168594f80039336501&number=1594

Not sure how that feeds back into the PR checks, it you look at the status checks here it seems to think the chromatic deployment hasn't even happened.

**Comment by ob6160** [comment]
> I think there are changes that need reviewing in Chromatic
Ah yes I missed that. I'll review those once the rebuild happens.

> Not sure how that feeds back into the PR checks
We've limited the Chromatic and e2e test GH action runs so they only happen when a review is requested. If you tag the identity team as a review on the PR it should trigger a rebuild.

We did this because doing a full run for each pushed commit had quite a hit on our Chromatic/Mailosaur usage.

**Pull request merged**